.github/workflows/release.yml

```
@@ -0,0 +1,111 @@
+        #   arch: x86
+        #   msvc_arch: x86
+        # - os: windows-2019
+        #   arch: AMD64
+        #   msvc_arch: x64
+        - os: macos-13
+          arch: x86_64
+          cmake_osx_architectures: x86_64
+        - os: macos-14
+          arch: arm64
+          cmake_osx_architectures: arm64
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Cache 3rd-party install directory
+        id: cache-build
+        uses: actions/cache@v4
+        with:
+          path: ${{ runner.temp }}/3rd-party/dist
+          key: ${{ matrix.os }}-${{ matrix.arch }}-${{ env.ABSL_VERSION }}-${{ env.S2GEOMETRY_VERSION }}-${{ env.S2GEOGRAPHY_VERSION }}-${{ hashFiles('ci/*') }}
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.20.0
+        env:
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_SKIP: cp36-* pp* *musllinux* *-manylinux_i686
+          CIBW_TEST_SKIP: "cp38-macosx_arm64"
+          CIBW_ENVIRONMENT_LINUX:
+            DEPENDENCIES_DIR=/host${{ runner.temp }}/3rd-party
+            CMAKE_PREFIX_PATH=/host${{ runner.temp }}/3rd-party/dist
+            ABSL_VERSION=${{ env.ABSL_VERSION }}
+            S2GEOMETRY_VERSION=${{ env.S2GEOMETRY_VERSION }}
+            S2GEOGRAPHY_VERSION=${{ env.S2GEOGRAPHY_VERSION }}
+            CXX_STANDARD=${{ env.CXX_STANDARD }}
+          CIBW_ENVIRONMENT_MACOS:
+            PROJECT_DIR=${{ github.workspace }}
+            DEPENDENCIES_DIR=${{ runner.temp }}/3rd-party
+            CMAKE_PREFIX_PATH=${{ runner.temp }}/3rd-party/dist
+            ABSL_VERSION=${{ env.ABSL_VERSION }}
+            S2GEOMETRY_VERSION=${{ env.S2GEOMETRY_VERSION }}
+            S2GEOGRAPHY_VERSION=${{ env.S2GEOGRAPHY_VERSION }}
+            CXX_STANDARD=${{ env.CXX_STANDARD }}
+            MACOSX_DEPLOYMENT_TARGET=10.9
+            CMAKE_OSX_ARCHITECTURES='${{ matrix.cmake_osx_architectures }}'
+          CIBW_ENVIRONMENT_WINDOWS:
+            DEPENDENCIES_DIR='${{ runner.temp }}\3rd-party'
+            ABSL_VERSION=${{ env.ABSL_VERSION }}
+            S2GEOMETRY_VERSION=${{ env.S2GEOMETRY_VERSION }}
+            S2GEOGRAPHY_VERSION=${{ env.S2GEOGRAPHY_VERSION }}
+            CXX_STANDARD=${{ env.CXX_STANDARD }}
+          CIBW_BEFORE_ALL: ./ci/install_3rdparty.sh
+          CIBW_BEFORE_ALL_WINDOWS: ci\install_3rdparty.cmd
+          # CIBW_BEFORE_BUILD_WINDOWS: pip install delvewheel
+          CIBW_REPAIR_WHEEL_COMMAND_LINUX: 'LD_LIBRARY_PATH=/host${{ runner.temp }}/3rd-party/dist/lib64 auditwheel repair -w {dest_dir} {wheel}'
+          CIBW_REPAIR_WHEEL_COMMAND_MACOS: 'LD_LIBRARY_PATH=${{ runner.temp }}/3rd-party/dist/lib delocate-wheel --require-archs=${{ matrix.arch }} -w {dest_dir} -v {wheel}'
+          # CIBW_REPAIR_WHEEL_COMMAND_WINDOWS: delvewheel repair --add-path ${{ runner.temp }}\geos-${{ env.GEOS_VERSION }}\bin -w {dest_dir} {wheel}
+          CIBW_TEST_REQUIRES: pytest
+          CIBW_TEST_COMMAND: pytest spherely
```

```suggestion
          CIBW_TEST_COMMAND: pytest --pyargs spherely
```

---

Hmm are the tests files installed with the spherely package?

---

Ah, no, not necessarily (probably not, actually, since they don't live inside the package). Didn't think about that, for shapely they are included and I know we struggled getting the correct `pytest` invocation in the cibuildwheel setup

---

```suggestion
          CIBW_TEST_COMMAND: pytest tests
```

then ? (assuming we are in the cloned repo)

---

Yes I guess we would either need to include the tests in the installed package or split into two different commands for linux (container) vs. macos so we can set the right path to the tests directory. Including tests may be better I guess.

---

Why would the path be different? When we are in the docker container, the cwd is not the repo source?

---

(given `./ci/install_3rdparty.sh` works to install the dependencies both on linux and mac, that makes me assume that `pytest tests/` should also work on both?)

---

Yes that's what I set in the last commit, let's see

---

OK, so I was wrong :(

```
  ERROR: file or directory not found: tests
```